cvr/src/main/java/org/cbioportal/cmo/pipelines/cvr/consume/ConsumeSampleWriter.java

```diff
@@ -116,44 +127,53 @@ public void write(List<? extends String> sampleIdList) throws Exception {
         }
     }
 
+    private void logConsumeSampleFailure(String sampleId, int numberOfRequestsAttempted, String message) {
+        log.error(String.format("Error consuming sample %s (after %d attempts) %s", sampleId, numberOfRequestsAttempted, message));
+    }
+
     private void consumeSample(String sampleId) {
-        HttpEntity requestEntity = getRequestEntity();
-        RestTemplate restTemplate = new RestTemplate();
-        SimpleClientHttpRequestFactory requestFactory =
-            (SimpleClientHttpRequestFactory) restTemplate.getRequestFactory();
-        requestFactory.setReadTimeout(10000);
-        requestFactory.setConnectTimeout(10000);
-        ResponseEntity<CVRConsumeSample> responseEntity;
-        try {
-            responseEntity = restTemplate.exchange(dmpConsumeUrl + sampleId, HttpMethod.GET, requestEntity, CVRConsumeSample.class);
-            if (responseEntity.getBody().getaffectedRows()==0) {
-                String message = "No consumption for sample " + sampleId;
-                log.warn(message);
-            }
-            if (responseEntity.getBody().getaffectedRows()>1) {
-                String message = "Multiple samples consumed (" + responseEntity.getBody().getaffectedRows() + ") for sample " + sampleId;
-                log.warn(message);
-            }
-            if (responseEntity.getBody().getaffectedRows()==1) {
-                String message = "Sample "+ sampleId +" consumed succesfully";
-                log.info(message);
-                // skip adding gml samples to this list since it is for smile only
-                // and smile is not taking in gml sample metadata
-                if (!gmlMode) {
-                    cvrSampleListUtil.updateSmileSamplesToPublishList(sampleId);
+        HttpEntity<LinkedMultiValueMap<String,Object>> requestEntity = getRequestEntity();
+        HttpClientWithTimeoutAndRetry client = new HttpClientWithTimeoutAndRetry(
+                dmpConsumeInitialResponseTimeout,
+                dmpConsumeMaximumResponseTimeout,
+                InstantStringUtil.createInstant(dropDeadInstantString),
+                false); // on a server error response, stop trying and fail/log (but continue on to other samples)
+        ResponseEntity<CVRConsumeSample> responseEntity = client.exchange(dmpConsumeUrl + sampleId, HttpMethod.GET, requestEntity, null, CVRConsumeSample.class);
+        if (responseEntity == null) {
+            String message = "";
+            if (client.getLastResponseBodyStringAfterException() != null) {
+                message = String.format("final response body was: '%s'", client.getLastResponseBodyStringAfterException());
+            } else {
```

Comment:
`else if () {`

Reply:
following google java format, the syntactically optional braces for blocks after "if" "else" "for" "do" and "while" are always to be used:

https://google.github.io/styleguide/javaguide.html#s4.1.1-braces-always-used